src/lib.rs

```diff
@@ -15,28 +16,35 @@ pub use session::{RequestSession, SessionMiddleware};
 
 mod session;
 
-pub struct Middleware {
-    key: Vec<u8>,
-}
+#[derive(Default)]
+pub struct Middleware {}
 
 impl Middleware {
-    pub fn new(key: &[u8]) -> Middleware {
-        Middleware { key: key.to_vec() }
+    pub fn new() -> Self {
+        Default::default()
     }
 }
 
+fn parse_pair(key_value: &str) -> Option<(String, String)> {
```

Comment:
This function was created to distinguish [Cookie header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie) from  [Set-Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie).